### DISCUSSION TITLE
Unbale to install Airflow on WSL2.

### DISCUSSION THREAD

**Discussion by OSIVZ** [apache/airflow#26524] [Q&A]
My setup:
**WSL2 (UBUNTU).
Python: 3.8.x**
Following an official step to install by running 
_pip install 'apache-airflow==2.4.0' --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-2.4.0/constraints-3.8.txt",_ 
I'm getting the error.

_WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7fda8100a820>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /apache/airflow/constraints-2.4.0/constraints-3.8.txt_

_ERROR: Could not install packages due to an EnvironmentError: HTTPSConnectionPool(host='raw.githubusercontent.com', port=443): Max retries exceeded with url: /apache/airflow/constraints-2.4.0/constraints-3.8.txt (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7fda8100a8e0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution'))_

Network team has ensured me that firewall allows two way communication (in and out) from my desktop IP. 
I thought that it could be just temp glitch on github repository, however after 2 days of trying, the issue persists.

Any suggestions?
Thank you
OSIVZ

**Comment by Taragolis** [comment]
Unfortunately WSL2 has quite a few network issues which randomly appear on some Windows machines and on other never happen.
Most popular is [DNS Resolution](https://github.com/microsoft/WSL/issues/8365) this kind of issue open a lot of times and always have been closed with something like "Can't be reproduce in Microsoft test environment.". 

You could try ping github by host name and by one of the ip address:
```python
❯ ping github.com -c 5
❯ ping 140.82.121.4 -c 5
```

If you unable ping it by host name and no problem ping by IP that is definitely DNS Resolution issue.
But anyway this is related to WSL2 not Airflow itself.
You could try test your luck and try one of the solutions from the issue and related issues which I mention above.

**Reply by OSIVZ** [reply]
Thank you for the quick response. 

**ping github.com -c 5** 
Output:
**ping: github.com: Temporary failure in name resolution**

However:
**ping 140.82.121.4 -c 5**
Output:
**PING 140.82.121.4 (140.82.121.4) 56(84) bytes of data.
64 bytes from 140.82.121.4: icmp_seq=1 ttl=47 time=92.4 ms
64 bytes from 140.82.121.4: icmp_seq=2 ttl=48 time=92.5 ms
64 bytes from 140.82.121.4: icmp_seq=3 ttl=47 time=92.4 ms
64 bytes from 140.82.121.4: icmp_seq=4 ttl=48 time=92.6 ms
64 bytes from 140.82.121.4: icmp_seq=5 ttl=47 time=92.9 ms**

I tried to use IP address  instead of references to "github" in pip command, but it resulted same issue as before. 
Let me read DNS Resolution blog.
Thank you
OSIVZ

**Reply by potiuk** [reply]
You also might have environmental problem (proxy/firewall etc). But for sure it is an environmental problem I am afraid.

**Reply by Taragolis** [reply]
Yeah it might be some environment problem but to be honest usual it some unknown problem (problem known stable solution no 🙄)  with WSL2

Luckily it doesn't happen that often personally I did not have that issue but I'm not usual use Windows and WSL2 on daily basis only when need to create some documentation related to Windows Host / WSL. However my team members and customers employees have.

Usual symptoms:
- DNS resolution doesn't work in WSL2
- Windows Host operate fine
- WSL v1 could resolve host name

Funnily enough all cases that I know solved by different solution provided from the Internet and WSL github.

**Reply by potiuk** [reply]
I think the REAL problem is WSL2 itself :rofl: 

We are quite close to a native support for Windows for both - development and running Airflow for testing (https://github.com/apache/airflow/issues/10388) and I think THIS is the right answer :)
